### ISSUE TITLE
rework MainIndexService assumptions

### ISSUE THREAD

**Issue by julianseeger** [Qabel/qabel-core#615]
The MainIndexService makes some assumptions that don't make sense from the clients perspective, those should be checked and fixed.

For example:
Removing the identity is only possible with an email or phonenumber that is neither null nor blank. So I can't delete a previously uploaded identity when I previously deleted those fields ... but a user that no longer wants to be found would probably do exactly that and thus, could never delete his identity using the MainIndexService.
The scenarios get much worse if the identity gets deleted and re-imported...

**Comment by enkore** [comment]
Yeah, it would be easier if we just allowed to say "remove *everything* you have on this identity". (This is server side.)

**Comment by enkore** [comment]
Scope:

- Review MIS assumptions
- Align MIS model with server model
- Depends qabel/qabel-index#43 and qabel/qabel-index#44

**Comment by thechauffeur** [comment]
Let us talk about this next time.
